document-generator-prosecution/src/main/java/uk/gov/companieshouse/document/generator/prosecution/ProsecutionDocumentInfoService.java

```diff
@@ -86,103 +60,21 @@ public ProsecutionDocumentInfoService(
+        final Map<String, Object> debugMap = new HashMap<>();
         debugMap.put("resource_uri", resourceUri);
-        LOG.infoContext(requestId,"Started getting document info", debugMap);
+        LOG.infoContext(requestId, "Started getting document info for prosecution", debugMap);
 
         String docGenUri = documentInfoRequest.getResourceUri();
-        if (isProsecutionUltimatumRequest(docGenUri)) {
-            return getUltimatumInfo(docGenUri, requestId);
-        } else if (isProsecutionSjpRequest(docGenUri)) {
-            return getSjpInfo(docGenUri, requestId);
-        } else {
-            Map<String, Object> logMap = new HashMap<>();
-            logMap.put(LOGGING_RESOURCE_KEY, docGenUri);
-            LOG.error("Unmatchable resourceUri inside prosecution: " + docGenUri, logMap);
-            throw new DocumentInfoException(
-                            "Unmatchable resourceUri inside prosecution: " + docGenUri);
-        }
-    }
-
-    private boolean isProsecutionUltimatumRequest(String urlToMatch) {
-        return urlToMatch.startsWith(MATCH_START_OF_ULTIMATUM_URI);
-    }
-
-    private boolean isProsecutionSjpRequest(String urlToMatch) {
-        return urlToMatch.startsWith(MATCH_START_OF_SJP_URI);
-    }
-
-    private String getProsecutionCaseUriFromUltimatumUri(String prosecutionCaseUri) {
-        return prosecutionCaseUri.substring(ULTIMATUM_INDEX_TO_TRIM_FROM);
-    }
-
-    private String getProsecutionCaseUriFromSjpUri(String docGenUri) {
-        return docGenUri.substring(SJP_INDEX_TO_TRIM_FROM);
-    }
-
-    private ProsecutionCase getProsecutionCase(String docGenUri, String prosecutionCaseUri,
-                    String requestId) throws DocumentInfoException {
+        ProsecutionDocument document;
         try {
-            return prosecutionClient.getProsecutionCase(prosecutionCaseUri, requestId);
-        } catch (SdkException ex) {
-            Map<String, Object> logMap = new HashMap<>();
-            logMap.put(LOGGING_RESOURCE_KEY, docGenUri);
-            LOG.errorContext(requestId,
-                            "Could not get prosecution case info to build template for URI: "
-                                            + prosecutionCaseUri,
-                            ex, logMap);
-            LOG.errorContext(requestId,
-                            "Could not get prosecution case info to build template for URI, cause: "
-                                            + prosecutionCaseUri,
-                            (Exception) ex.getCause(), logMap);
-            throw new DocumentInfoException("Could not build template: " + docGenUri, ex);
+            document = handler.getProsecutionDocument(resourceUri);
+            return handler.getDocumentResponse(document, requestId, document.getProsecutionCase().getStatus());
+        } catch (HandlerException e) {
+            LOG.error("Unmatchable resourceUri inside prosecution request : " + docGenUri);
```

Comment:
The handler exception is thrown for multiple different reasons. This error message could be misleading in some circumstances. It could be left generic.

```
Map<String, Object> logData = new HashMap<>();
logData.put("resource_uri", uri);
LOG.errorContext(requestId, "Failed to generate prosecution document", e, logData);
```

Reply:
Okay, done